src/sidebar/search/AddressInput.tsx

```diff
@@ -15,43 +15,75 @@ export interface AddressInputProps {
 }
 
 export default function AddressInput(props: AddressInputProps) {
-    const searchInput = useRef<HTMLInputElement>(null)
-
-    // holds the query text making this a controlled component
+    // controlled component pattern with initial value set from props
     const [text, setText] = useState(props.point.queryText)
-    // puts the query text from props into the state of this component
     useEffect(() => setText(props.point.queryText), [props.point.queryText])
 
+    // container for geocoding results which get set by the geocoder class and set to empty if the undelying query point gets changed from outside
     const [geocodingResults, setGeocodingResults] = useState<GeocodingHit[]>([])
     const [geocoder] = useState(new Geocoder(hits => setGeocodingResults(hits)))
-    useEffect(() => setGeocodingResults([]), [props.point])
+    useEffect(() => {
+        setGeocodingResults([])
+        //setFullscreen(false)
+    }, [props.point])
 
-    const [fullscreen, setFullscreen] = useState(false)
+    // highlighted result of geocoding results. Keep track which index is highlighted and change things on ArrowUp and Down
+    // on Enter select highlighted result or the 0th if nothing is highlighted
+    const [highlightedResult, setHighlightedResult] = useState<number>(-1)
+    useEffect(() => setHighlightedResult(-1), [geocodingResults])
+    const searchInput = useRef<HTMLInputElement>(null)
+    const onKeypress = useCallback(
+        (event: React.KeyboardEvent<HTMLInputElement>) => {
+            if (event.key === 'Escape') {
+                searchInput.current!.blur()
+                return
+            }
+            if (geocodingResults.length === 0) return
+
+            switch (event.key) {
+                case 'ArrowUp':
+                    setHighlightedResult(i => calculateHighlightedIndex(geocodingResults.length, i, -1))
+                    break
+                case 'ArrowDown':
+                    setHighlightedResult(i => calculateHighlightedIndex(geocodingResults.length, i, 1))
+                    break
+                case 'Enter':
+                    // by default use the first result, otherwise the highlighted one
+                    const index = highlightedResult >= 0 ? highlightedResult : 0
+                    props.onAddressSelected(geocodingResults[index])
+                    searchInput.current!.blur()
+                    break
+            }
+        },
+        [geocodingResults, highlightedResult]
+    )
 
+    // toggle fullscreen display on small screens
+    const [fullscreen, setFullscreen] = useState(false)
     const containerClass = fullscreen ? styles.container + ' ' + styles.fullscreen : styles.container
+
     return (
         <div className={containerClass}>
             <div className={styles.inputContainer}>
                 <input
                     className={styles.input}
-                    type='text'
+                    type="text"
```

Comment:
maybe the code formatting settings were off here?

Reply:
I think in HTML-Tags, the formatting uses ". Everywhere else ' is used.

Reply:
Hm, when I run 'Reformat Code' in IntelliJ it changes the quotes to single quotes

Reply:
Ugh sorry, I did not know 'Reformat Code' and 'Reformat with Prettier' are two different commands even after enabling the Prettier code style: https://intellij-support.jetbrains.com/hc/en-us/community/posts/360002745119-How-to-have-Reformat-Code-feature-behave-exactly-like-Prettier-

Using `Reformat with Prettier` the quotes are actually changed to double quotes, unless I add ` "jsxSingleQuote": true` to `prettierrc`.

Reply:
Oh no! 🙄